### ISSUE TITLE
mnist-7.onnx and mnist-8.onnx accuracy regression when using NNPA

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx-mlir#1453]
# Description
When using our custom model client, we noticed an accuracy regression from the mnist-7 and 8 models from https://github.com/onnx/models/tree/main/vision/classification/mnist/model (we don't run mnist-1)

The left side is onnx-mlir 0bbf9ec73ebfecf3a26d7e3791e267b4bab5276d and the right side is 5967ddc3701a7bd4845d14566c98e006e821dc67. The "actual" values are the ones we get after running the model compiled from onnx-mlir. The "expected" values come from the onnx-modelzoo itself ([mnist-8 data set 0](https://github.com/onnx/models/blob/main/vision/classification/mnist/model/mnist-8.tar.gz)).
Because NNPA uses reduced precision, we don't expect the actual vs expected values to perfectly match. The issue though is why it appears the values are being rounded now. 
<img width="683" alt="image" src="https://user-images.githubusercontent.com/42243335/169592395-ae455dc2-6ff8-497f-8c40-df18ffeb4461.png">

Notes:
* Built onnx-mlir using "Release" setting and enabled NNPA acceleration
* Built model using `onnx-mlir --O3 --EmitLib --mtriple=s390x-ibm-loz --mcpu=z14 --maccel=NNPA mnist-8.onnx -v`

**Comment by messerb5467** [comment]
The great thing looking at this is this won't statistically affect the output of any models compiled with onnx-mlir since the values are so close. That being said, I suspect there's still a recent commit that probably affected this and if we can find it, I expect we could arrive at our answer very quickly.

**Comment by cjvolzka** [comment]
Yeah, if the values just adjusted a little, the difference isn't enough to cause a lot of concern (ie the model still classifies all the images the same way). The thing that's concerning me is that all but one of them have basically become integers. That strikes me as quite odd and might be a sign of something more is happening.

**Comment by messerb5467** [comment]
Agreed on both accounts. We should find what happened with the code somewhere.
